src/shogun/machine/Machine.h

```diff
@@ -315,7 +329,6 @@ class CMachine : public CStoppableSGObject
 
 		virtual const char* get_name() const { return "Machine"; }
 
-	protected:
```

Comment:
you want `train_machine` to be public? why?

Reply:
this is a mistake...  no need for public ;)